extensions/scarb-doc/src/types.rs

```diff
@@ -214,15 +252,51 @@ pub struct Struct {
     pub id: StructId,
     pub node: ast::ItemStructPtr,
 
+    pub members: Vec<Member>,
+
     pub item_data: ItemData,
+    pub lookup_item_data: LookupItemData,
 }
 
 impl Struct {
-    pub fn new(db: &dyn DefsGroup, id: StructId, node: &ast::ItemStruct) -> Self {
+    pub fn new(db: &RootDatabase, id: StructId) -> Self {
+        let members = db.struct_members(id).unwrap();
+        let members = members
+            .iter()
+            .map(|(_name, semantic_member)| Member::new(db, semantic_member.id))
+            .collect::<Vec<_>>();
+
+        let node = id.stable_ptr(db);
+        Self {
+            id,
+            node,
+            members,
+            item_data: ItemData::new(db, id, node),
+            lookup_item_data: LookupItemData::new(
+                db,
+                LookupItemId::ModuleItem(ModuleItemId::Struct(id)),
+            ),
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct Member {
+    pub id: MemberId,
+    pub node: ast::MemberPtr,
+
+    pub item_data: ItemData,
+    // TODO: no LookupItemId enum variant for struct / enum items
```

Comment:
I kind of worry that we'll lose the track of those

Reply:
Fixed